apps/src/args.rs

```diff
@@ -273,6 +281,8 @@ Options:
   --max-active-cids NUM    The maximum number of active Connection IDs we can support [default: 2].
   --enable-active-migration   Enable active connection migration.
   --perform-migration      Perform connection migration on another source port.
+  --multipath              Enable multipath support.
+  -A --address ADDR ...    Additional client addresses to use.
```

Comment:
This description is kind of confusing. 

Prior to this change, the client would look at the Server IP and pick an "any" socket using an IP family that matched the server's i.e.  0.0.0.0:0 or [::]:0 depending on v4 or v6 respectively

With this change, there aren't _additional_ address that are used on top of the old behaviour, but instead they are just the addresses that would be used. This opens a few new failure scenarios that could catch users out. For example, if a server only returns an A record and the user provided a v6 client address, then the connection would fail. 


Among the most confusing type of failure is where packets go to a black hole and the connection fails after a timeout.

We might want to tweak how the socket code handling for this option works, and make the description a bit clearer about what it does and how it might fail.

Reply:
You're right, the current description does not suit the actual behavior of this option. As a first step, the initial description could be rewritten as "Specify source addresses to be used instead of the unspecified address" (or "instead of letting the OS decides",...).

For the family address mismatch that may arise, I can indeed rework the code to remove all the specified addresses that do not match the family of the server one. In the case none is remaining, the code can hence fallback to the original behavior, i.e., use "0.0.0.0:0" or "[::]:0".

There still remains the issue that non-routable addresses may be provided, e.g., `fe80::/16` or other. I'm not sure we can do much here, so maybe update the description as "Specify source addresses to be used instead of the unspecified address, non-routable addresses will lead to connectivity issues" (maybe a bit long, but not sure how to make it shorter without loosing information).